src/main/java/io/jsonwebtoken/JwtParser.java

```diff
@@ -338,6 +358,60 @@
     <T> T parse(String jwt, JwtHandler<T> handler)
             throws ExpiredJwtException, UnsupportedJwtException, MalformedJwtException, SignatureException, IllegalArgumentException;
 
+    
+    /**
+     * Parses the header part of the specified compact serialized JWT string.
+     * 
+     * <p>Note that while parsing header no validation or security check will be applied.</p>
+     * 
+     * @param jwt the compact serialized JWT to parse
+     * @return Deserialized {@link Header} part of specified compact JWT string.
+     */
+	Header parseHeader(String jwt);
+
+	/**
+	 * Checks if the specified compact serialized JWT string is expired based on the builder's current configuration state. 
+     * <p>
+     * <p>Note that you need this method probably only then, if the parser is configured to by pass 'expiration time' ({@code exp}) claim validation (by calling builder's {@link #ignoreExpiry()} method).</p>
+     *
+     * @param jwt the compact serialized JWT to check
+     * @return {@code true} if the specified JWT compact string is expired, {@code false} otherwise.
+	 */
+	boolean isExpired(String jwt);
+	
+	/**
+	 * Checks if the specified compact serialized JWT string is premature based on the builder's current configuration state. 
+     * <p>
+     * <p>Note that you need this method probably only then, if the parser is configured to by pass 'not before' ({@code nbf}) claim validation (by calling builder's {@link #ignoreNotBefore()} method).</p>
+     *
+     * @param jwt the compact serialized JWT to check
+     * @return {@code true} if the specified JWT compact string is premature, {@code false} otherwise.
+	 */
+	boolean isPremature(String jwt);
+	
+	
+	/**
+	 * Validates 'expiration time' ({@code exp}) claim of the specified compact serialized JWT string based on the builder's current configuration state. 
+     * <p>
+     * <p>Note that you need this validation method probably only then, if the parser is configured to by pass 'expiration time' ({@code exp}) claim validation (by calling builder's {@link #ignoreExpiry()} method).</p>
+     *
+     * @param jwt the compact serialized JWT to check
+     * 
+     * @throws ExpiredJwtException if the validation of 'expiration time' ({@code exp}) claim is failed.
+	 */
+	void validateExpiration(String jwt);
```

Comment:
isExpired method is just a wrapper on validateExpiration. I explained already above in my first comment, why we need isExpired method. So in my case again we can let validateExpiration method private. But _again_ why to remove this method? Both validateExpiration and validateNotBefore methods was in the code already, just there was a part of the huge parse method. I only moved that parts of the code into the smaller methods and made them accessible. So the decision is up to you.